packages/perseus-core/src/widgets/matcher/matcher-util.ts

```diff
@@ -0,0 +1,94 @@
+import {CoreUtil} from "@khanacademy/perseus-core";
+
+import type {PerseusMatcherWidgetOptions} from "@khanacademy/perseus-core";
+
+// TODO(LEMS-2841): Should be able to remove once getPublicWidgetOptions is hooked up
+type MatcherInfo = {
+    left: ReadonlyArray<string>;
+    right: ReadonlyArray<string>;
+    orderMatters: boolean;
+    problemNum: number | null | undefined;
+};
+
+type MatcherShuffleInfo = {
+    left: ReadonlyArray<string>;
+    right: ReadonlyArray<string>;
+    orderMatters: boolean;
+};
+
+// TODO(LEMS-2841): Should be able to remove once getPublicWidgetOptions is hooked up
+export const shuffleMatcher = (
+    props: MatcherInfo,
+): {left: ReadonlyArray<string>; right: ReadonlyArray<string>} => {
+    // Use the same random() function to shuffle both columns sequentially
+    const rng = CoreUtil.seededRNG(props.problemNum as number);
+
+    let left;
+    if (!props.orderMatters) {
+        // If the order doesn't matter, don't shuffle the left column
+        left = props.left;
+    } else {
+        left = CoreUtil.shuffle(props.left, rng, /* ensurePermuted */ true);
+    }
+
+    const right = CoreUtil.shuffle(props.right, rng, /* ensurePermuted */ true);
+
+    return {left, right};
+};
+
+// TODO(LEMS-2841): Can shorten to shuffleMatcher after above function removed
+function shuffleMatcherWithRandom(data: MatcherShuffleInfo): {
+    left: ReadonlyArray<string>;
+    right: ReadonlyArray<string>;
+} {
+    // Use the same random() function to shuffle both columns sequentially
+    let left;
+    if (!data.orderMatters) {
+        // If the order doesn't matter, don't shuffle the left column
+        left = data.left;
+    } else {
+        left = CoreUtil.shuffle(
+            data.left,
+            Math.random() * 100,
+            /* ensurePermuted */ true,
+        );
+    }
+
+    const right = CoreUtil.shuffle(
+        data.right,
+        Math.random() * 100,
```

Comment:
Multiplied the Math.random() here by 100 to get the shuffle to be random each time. Without multiplying, the shuffle is the same each time, though it is different than the answer. I was expecting it to be random, so updated to get that response.

Reply:
Would it be worth making this a comment in the code?

Reply:
Ben pointed out why this is happening and how I just need to pass in Math.random instead, so will update that here.

Reply:
Updated :)